components/uniset/src/uniset.rs

```diff
@@ -0,0 +1,366 @@
+use std::{boxed::Box, error::Error, iter::Iterator, slice::Iter, vec::Vec};
+
+/// Represents the maximum Unicode Code Point, inclusive
+const CODEPOINT_MAX: u32 = 0x10FFFF;
```

Comment:
Can you use https://doc.rust-lang.org/std/char/constant.MAX.html ?

Reply:
We can, but there doesn't seem to be a MIN constant.  I am unsure if having a MIN constant is too important because of the nature of u32 being unsigned. If we are to define a MIN constant, I feel that defining a separate MAX constant can allow for some clarity when looking through the code.

Reply:
MIN is trivially zero.

If you want to keep the explicit `0x10FFFF`, you could use `const_assert!` that it is equal to `std::char::MAX` to help people reading the code.

Reply:
agreed and changed (just sticking to `std::char::MAX`)